File: system-test/dotnet/Driver.Adapter/External/Tax/TaxRealDriver.cs
using Common;
using Driver.Adapter.External.Tax.Client;
using Driver.Port.External.Tax.Dtos;
using Driver.Port.External.Tax.Dtos.Error;

namespace Driver.Adapter.External.Tax;

public class TaxRealDriver : BaseTaxDriver<TaxRealClient>
{
    public TaxRealDriver(string baseUrl) : base(new TaxRealClient(baseUrl))
    {
    }

    public override Task<Result<VoidValue, TaxErrorResponse>> ReturnsTaxRateAsync(ReturnsTaxRateRequest request)
    {
        // No-op for real driver - data already exists in real service
        return Task.FromResult(Result.Success<TaxErrorResponse>());
    }
}

File: system-test/java/src/main/java/com/mycompany/myshop/testkit/driver/adapter/external/tax/TaxRealDriver.java
package com.mycompany.myshop.testkit.driver.adapter.external.tax;

import com.mycompany.myshop.testkit.driver.adapter.external.tax.client.TaxRealClient;
import com.mycompany.myshop.testkit.driver.port.external.tax.dtos.ReturnsTaxRateRequest;
import com.mycompany.myshop.testkit.driver.port.external.tax.dtos.error.TaxErrorResponse;
import com.mycompany.myshop.testkit.common.Result;

public class TaxRealDriver extends BaseTaxDriver<TaxRealClient> {
    public TaxRealDriver(String baseUrl) {
        super(new TaxRealClient(baseUrl));
    }

    @Override
    public Result<Void, TaxErrorResponse> returnsTaxRate(ReturnsTaxRateRequest request) {
        return Result.success();
    }
}
